src/Command/Commands.js

```diff
@@ -7,11 +7,13 @@ module.exports = Command.extend({
     dependencies: {
         'commandList': 'commandList'
     },
-    processMessage: function (message, tokens) {
+    processMessage: function (message) {
         let commands = this.commandList.all();
         let advertisableList = [];
         for (let key in commands) {
-            if (commands[key].advertisable) advertisableList.push("`!" + commands[key].commandName + "`");
+            if (commands.hasOwnProperty(key) && commands[key].advertisable) {
+                advertisableList.push(`\`!${key}\``);
```

Comment:
All keys are now equivalent to the command name (or alias). It used to be the file name, but since this is no longer the case we can simply write down the key here.

Reply:
Along with my suggestion to use two command lists, I would revert this modification.

Reply:
I can revert it, but it makes no difference in implementation. The keys on the command list are the commandNames for each command (which is necessary to look up commands). Using these keys to fetch the command, to then fetch the commandName (commandName === key) is extra work for no reason.